core/server/proxy/src/test/java/alluxio/StreamCacheTest.java

```diff
@@ -47,70 +48,127 @@ public void concurrentOperations() throws Exception {
     final StreamCache streamCache = new StreamCache(Constants.HOUR_MS);
     final FileInStream is = Mockito.mock(FileInStream.class);
     final FileOutStream os = Mockito.mock(FileOutStream.class);
+    final int numThreads = 100;
 
     // Concurrent put.
-    final AtomicInteger isId = new AtomicInteger(0);
-    Runnable putIs = new Runnable() {
+    final AtomicIntegerArray isIDs = new AtomicIntegerArray(numThreads);
+    final AtomicInteger isIDsIndex = new AtomicInteger();
+    final AtomicIntegerArray osIDs = new AtomicIntegerArray(numThreads);
+    final AtomicInteger osIDsIndex = new AtomicInteger();
+    Runnable put = new Runnable() {
       @Override
       public void run() {
-        isId.set(streamCache.put(is));
+        int id = streamCache.put(is);
+        int index = isIDsIndex.getAndIncrement();
+        isIDs.set(index, id);
+        id = streamCache.put(os);
```

Comment:
nit: I would prefer to have separate threads for the input stream and output stream puts